### PR TITLE
[Data][Docs] Revise user guides

### PR DESCRIPTION
**Pull request by bveeramani**
**Stacked on**:
- [ ] #35916 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/saving-data.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            import ray
+
+            ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+
+            ds.write_csv("/tmp/iris/")
+
+
+    .. tab-item:: GCP
+
+        ham
```
Comment (amogkam) [inline doc/source/data/saving-data.rst:58]: add azure?

File: doc/source/data/saving-data.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    ds.write_csv("/mnt/cluster_storage/iris")
+
+To write data to formats other than CSV, read the :ref:`Input/Output reference <input-output>`.
+
+Changing the number of output files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you call a write method, Ray Data writes your data to one or more files at the
+specified path. To change the number of output files, call :meth:`~ray.data.Dataset.repartition`.
```
Comment (amogkam) [inline doc/source/data/saving-data.rst:82]: explicitly say 1 file per block (and link to inspecting data to know how many blocks there are)

File: doc/source/data/transforming-data.rst [hunk 3]
```diff
@@ -1,286 +1,227 @@
...
+.. tip::
 
-    from ray.data.aggregate import Mean, Std
+    :meth:`~ray.data.Dataset.random_shuffle` is slow. For better performance, try
+    :ref:`iterating-over-batches-with-shuffling`.
 
-    # Global mean on B and C columns.
-    ds.mean(["B", "C"])
-    # -> {'mean(B)': 9.0, 'mean(C)': 13.5}
+Creating rows
```
Comment (amogkam) [inline doc/source/data/transforming-data.rst:208]: I think creating rows is a bit of weird term. Both `map` and `flat_map` should be in "Transforming Rows" section. And then differentiate between the two by saying 1:1 is map, 1:N is flat_map

**Review by amogkam — COMMENTED** [review]
nice work! code snippets lgtm overall. left some comments

**Comment by bveeramani** [comment]
Closing in favor of the following PRs:
* https://github.com/ray-project/ray/pull/36121
* https://github.com/ray-project/ray/pull/36144
* https://github.com/ray-project/ray/pull/36162

**Pull request closed without merging**